File: client/src/main/scala/stasis/client/encryption/secrets/DeviceFileSecret.scala
package stasis.client.encryption.secrets

import org.apache.pekko.util.ByteString

import stasis.client.encryption.Aes
import stasis.client.encryption.stream.CipherStage

final case class DeviceFileSecret(
  file: String,
  iv: ByteString,
  private val key: ByteString
) extends Secret {
  def encryption: CipherStage = Aes.encryption(key, iv)
  def decryption: CipherStage = Aes.decryption(key, iv)
}

object DeviceFileSecret {
  def apply(file: String, iv: ByteString, key: ByteString): DeviceFileSecret =
    new DeviceFileSecret(file, iv, key)
}

File: client-android/lib/src/main/kotlin/stasis/client_android/lib/encryption/secrets/DeviceFileSecret.kt
package stasis.client_android.lib.encryption.secrets

import okio.ByteString
import okio.Sink
import okio.Source
import stasis.client_android.lib.encryption.Aes

data class DeviceFileSecret(
    val file: String,
    val iv: ByteString,
    private val key: ByteString
) : Secret() {
    fun encryption(source: Source): Source = Aes.encryption(source, key, iv)
    fun encryption(sink: Sink): Sink = Aes.encryption(sink, key, iv)
    fun decryption(source: Source): Source = Aes.decryption(source, key, iv)
}
